src/components/App.js

```diff
@@ -26,38 +26,35 @@ const SecondaryPanes = createFactory(require("./SecondaryPanes"));
 const WelcomeBox = createFactory(require("./WelcomeBox"));
 const EditorTabs = createFactory(require("./Editor/Tabs"));
 
-const App = React.createClass({
-  propTypes: {
-    sources: PropTypes.object,
-    selectSource: PropTypes.func,
-    selectedSource: PropTypes.object,
-    startPanelCollapsed: PropTypes.bool,
-    endPanelCollapsed: PropTypes.bool,
-  },
+class App extends React.Component {
+  state: {
+    horizontal: verticalLayoutBreakpoint.matches
+  }
 
-  displayName: "App",
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
Hmm yes, but I was getting some flow errors if I don't define state out of constructor
`Did you forget to declare type parameter `State` of property `Component`?`

Any suggestions?